redhat-developer/developers.redhat.com
# [RHDX-102] Multiple twig render array warnings

### JIRA Issue Link
[RHDX-102](https://projects.engineering.redhat.com/browse/RHDX-102)

Affects:
- https://localhost/articles/deploy-a-NodeJS-app-OpenShift/
- https://localhost/coderland/serverless/deploying-serverless-knative/
- https://localhost/coderland/serverless/building-a-serverless-service/
- https://localhost/che-custom-stacks/

PR environment URLs:
- https://developer-preview-3447.ext.us-west.dc.preprod.paas.redhat.com/articles/deploy-a-NodeJS-app-OpenShift/
- https://developer-preview-3447.ext.us-west.dc.preprod.paas.redhat.com/coderland/serverless/deploying-serverless-knative/
- https://developer-preview-3447.ext.us-west.dc.preprod.paas.redhat.com/coderland/serverless/building-a-serverless-service/
- https://developer-preview-3447.ext.us-west.dc.preprod.paas.redhat.com/che-custom-stacks/

### Verification Process

The only PHP warning that remains (Notice: Undefined offset: 0 in rhdp2_preprocess_assembly()) is related to a different issue, but the following warnings should no longer occur, for each usage of Bonus Content Item attached to node:

User error: "term_name" is an invalid render array key...
User error: "icon_style" is an invalid render array key…
User error: "term_image_url_alt" is an invalid render…
User error: "term_image_url" is an invalid render array...

The warnings are resolved by adding the #markup to the items attached to the $build.
The PHP was optimized for returning the $build only once.

### Testing Process
- View the content that currently uses BonusContent and verify the output has not changed
- Create test content for BonusContent  and a Bonus Content Item that does not have a 'Bonus content type' selected.  Verify the default icon/alt displays.
- Test a taxonomy term that doesn't have an image or icon:
-- Go to the Bonus Content Type vocabulary.
-- Create a test term. Do not provide an icon or image.
-- Update your test content to use the new term.
-- Verify the content renders with the default image/alt.
-- Delete the test taxonomy from the Vocabulary
-- Verify the page loads without any fatal errors
